src/test/java/org/opensearch/sdk/TestBaseExtension.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.junit.jupiter.api.Test;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.List;
+
+public class TestBaseExtension extends OpenSearchTestCase {
+
+    private static final String EXTENSION_DESCRIPTOR_CLASSPATH = "/bad-extension.yml";
+    private static final String EXTENSION_DESCRIPTOR_FILEPATH = "src/test/resources" + EXTENSION_DESCRIPTOR_CLASSPATH;
+
+    private class TestExtension extends BaseExtension {
+
+        protected TestExtension(String path) {
```

Comment:
Nit: seems silly to make this `protected` rather than public.

Reply:
You are right, I'll change it!